test/test.dart

```diff
@@ -318,24 +317,47 @@ extension WidgetTesterExtension on WidgetTester {
 
   Future<void> _screenMatchesGolden(
     String name, {
-    bool? autoHeight,
-    Finder? finder,
-    CustomPump? customPump,
-  }) {
+    Future<void> Function(WidgetTester)? customPump,
+  }) async {
     final testDescription = getTestDescription(
       lightTheme: lightThemeGolden,
       playerInterfaceColorStyle: nonDefaultPlayerInterfaceColorStyle,
     );
     name = testDescription.buildFileName(name);
-    return screenMatchesGolden(
-      this,
-      name,
-      autoHeight: autoHeight,
-      finder: finder,
-      customPump: customPump,
+    await _waitForAssets();
+    if (customPump != null) {
+      await customPump(this);
+    } else {
+      await pumpAndSettle();
+    }
+    await expectLater(
+      find.byWidgetPredicate((widget) => true).first, // The whole screen
+      matchesGoldenFile('goldens/$name.png'),
     );
   }
 
+  Future<void> _waitForAssets() async {
```

Comment:
Let's move this to a different file, maybe in some `golden_framework` folder next to loadAppFonts

Reply:
I extracted all golden test specific code and moved it in `golden/framework`.